beacon-chain/rpc/validator/server.go

```diff
@@ -62,7 +62,7 @@ type Server struct {
 	Eth1BlockFetcher       powchain.POWBlockFetcher
 	PendingDepositsFetcher depositcache.PendingDepositsFetcher
 	OperationNotifier      opfeed.Notifier
-	StateGen               *stategen.State
+	StateGen               stategen.StateManager
```

Comment:
For my benefit. why this change?

Reply:
Simply changes it from a struct to an interface, this allows us to mock it correctly for tests. `*stategen.State` fulfills the state manager interface.